streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -436,6 +436,15 @@ private void close(final boolean clean) {
         transitionTo(State.CLOSED);
     }
 
+    @Override
+    public void markChangelogAsCorrupted(final Set<TopicPartition> partitions) {
+        stateMgr.markChangelogAsCorrupted(partitions);
+
+        // only write a new checkpoint (excluding the corrupted partitions) if eos is disabled
```

Comment:
We can checkpoint (excluding the corrupted partitions) for 1) standby tasks and 2) non-eos active tasks, for eos active tasks we should not write the checkpoint since for eos we HAVE TO reboot every store from scratch to maintain consistency.